superset/views/core.py

```diff
@@ -449,10 +449,12 @@ def get_raw_results(self, viz_obj: BaseViz) -> FlaskResponse:
         payload = viz_obj.get_df_payload()
         if viz_obj.has_error(payload):
             return json_error_response(payload=payload, status=400)
-        return self.json_response({"data": payload["df"].to_dict("records")})
+        return self.json_response(
+            {"data": payload["df"].to_dict("records"), "colnames": payload["colnames"]}
```

Comment:
let's also add "coltypes" here so we get this feature working for legacy charts 👍

Reply:
Done!